test/e2e/common/node/container_probe.go

```diff
@@ -246,6 +246,25 @@ var _ = SIGDescribe("Probing container", func() {
 		runReadinessFailTest(f, pod, time.Minute)
 	})
 
+	/*
+		Release: v1.21
+		Testname: Pod liveness probe, container exec timeout, restart
+		Description: A Pod is created with liveness probe with a Exec action on the Pod. If the liveness probe call does not return within the timeout specified, liveness probe MUST restart the Pod. When ExecProbeTimeout feature gate is disabled, the timeout is ignored BUT a failing liveness probe MUST restart the Pod.
+	*/
+	ginkgo.It("should be restarted with a failing exec liveness probe that took longer than the timeout", func() {
+		// The ExecProbeTimeout feature gate exists to allow backwards-compatibility with pre-1.20 cluster behaviors, where livenessProbe timeouts were ignored
```

Comment:
You may have mentioned this already, so sorry for asking again. What's the reason for not conditionally running this test when ExecProbeTimeout=false? Is it because the test would pass regardless because of the timeout?